src/app/shared/components/validate-json-or-yaml/component.spec.ts

```diff
@@ -0,0 +1,230 @@
+
+  beforeEach(waitForAsync(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ValidateJsonOrYamlComponent],
+      imports: [MatFormFieldModule],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ValidateJsonOrYamlComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('Template', () => {
+    it('should render <mat-hint> when isModelValid is true', () => {
+      component.isDataValid = true;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element).toBeTruthy();
+    });
+
+    it('should render <mat-hint> when data is null', () => {
+      component.data = null;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element).toBeTruthy();
+    });
+
+    it('should render correct text inside <mat-hint> when isOnlyYAML is false', () => {
+      component.isDataValid = true;
+      component.isOnlyYAML = false;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element.textContent.trim()).toEqual('Input should be valid JSON or YAML');
+    });
+
+    it('should render correct text inside <mat-hint> when isOnlyYAML is true', () => {
+      component.isDataValid = true;
+      component.isOnlyYAML = true;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element.textContent.trim()).toEqual('Input should be valid YAML');
+    });
+
+    it('should render <mat-error> when data is truthy and isModelValid is false', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element).toBeFalsy();
+    });
+
+    it('should render <mat-error> when data is truthy and isModelValid is false', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-error');
+
+      expect(element).toBeTruthy();
+    });
+
+    it('should render correct text inside <mat-error> when isOnlyYAML is false', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      component.isOnlyYAML = false;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-error');
+
+      expect(element.textContent.trim()).toEqual('Please enter valid JSON or YAML');
+    });
+
+    it('should render correct text inside <mat-error> when isOnlyYAML is true', () => {
```

Comment:
should display correct error message when only validating YAML

Reply:
test name reworded